DREAM3DReviewFilters/GenerateFeatureIDsbyBoundingBoxes.cpp

```diff
@@ -0,0 +1,514 @@
+  std::vector<size_t> tDims(1, numFeatureIDs + 1);
+  m->createNonPrereqAttributeMatrix(this, getFeatureAttributeMatrixArrayPath(), tDims, m_DestAttributeMatrixType, AttributeMatrixID20);
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+bool IsPointInBounds(float xmax, float xmin, float ymax, float ymin, float zmax, float zmin, float x, float y, float z)
+{
+  bool inBounds = false;
+
+  if((x < xmax) && (x > xmin) && (y < ymax) && (y > ymin) && (z < zmax) && (z > zmin))
+  {
+    inBounds = true;
+  }
+
+  return inBounds;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void GenerateFeatureIDsbyBoundingBoxes::checkBoundingBoxImage()
+{
+  size_t totalNumFIDs = m_BoxFeatureIdsPtr.lock()->getNumberOfTuples();
+  size_t totalNumElementsDest = m_FeatureIdsPtr.lock()->getNumberOfTuples();
+  DataContainer::Pointer dc = getDataContainerArray()->getDataContainer(getFeatureIDsArrayPath().getDataContainerName());
+  ImageGeom::Pointer image = dc->getGeometryAs<ImageGeom>();
+  SizeVec3Type udims = image->getDimensions();
+  FloatVec3Type uorigin = image->getOrigin();
+  FloatVec3Type uspacing = image->getSpacing();
+
+  int64_t dims[3] = {
+      static_cast<int64_t>(udims[0]),
+      static_cast<int64_t>(udims[1]),
+      static_cast<int64_t>(udims[2]),
+  };
+
+  float origin[3] = {
+      static_cast<float>(uorigin[0]),
+      static_cast<float>(uorigin[1]),
+      static_cast<float>(uorigin[2]),
+  };
+
+  float spacing[3] = {
+      static_cast<float>(uspacing[0]),
+      static_cast<float>(uspacing[1]),
+      static_cast<float>(uspacing[2]),
+  };
+
+  bool inBounds = false;
+  float xmin = 0;
+  float xmax = 0;
+  float ymin = 0;
+  float ymax = 0;
+  float zmin = 0;
+  float zmax = 0;
+
+  float currentx = 0;
+  float currenty = 0;
+  float currentz = 0;
+
+  int64_t xindex = 0;
+  int64_t yindex = 0;
+  int64_t zindex = 0;
+  int64_t tempMod = 0;
+
+  size_t k = 0;
+
+  for(size_t i = 0; i < totalNumElementsDest; i++)
+  {
+
+    zindex = i / (dims[0] * dims[1]);
+    tempMod = i % (dims[0] * dims[1]);
+    yindex = tempMod / (dims[0]);
+    xindex = tempMod % (dims[0]);
+
+    currentx = origin[0] + spacing[0] * xindex;
+    currenty = origin[1] + spacing[1] * yindex;
+    currentz = origin[2] + spacing[2] * zindex;
+
+    inBounds = false;
+    k = 0;
+
+    while(!inBounds && k < totalNumFIDs)
+    {
+      xmin = m_BoxCenter[3 * k] - m_BoxDims[3 * k] / 2.0;
+      xmax = m_BoxCenter[3 * k] + m_BoxDims[3 * k] / 2.0;
+      ymin = m_BoxCenter[3 * k + 1] - m_BoxDims[3 * k + 1] / 2.0;
+      ymax = m_BoxCenter[3 * k + 1] + m_BoxDims[3 * k + 1] / 2.0;
+      zmin = m_BoxCenter[3 * k + 2] - m_BoxDims[3 * k + 2] / 2.0;
+      zmax = m_BoxCenter[3 * k + 2] + m_BoxDims[3 * k + 2] / 2.0;
```

Comment:
```suggestion
  for(size_t i = 0; i < totalNumElementsDest; i++)
  {
    int64_t zindex = i / (dims[0] * dims[1]);
    int64_t tempMod = i % (dims[0] * dims[1]);
    int64_t yindex = tempMod / (dims[0]);
    int64_t xindex = tempMod % (dims[0]);

    float currentx = origin[0] + spacing[0] * xindex;
    float currenty = origin[1] + spacing[1] * yindex;
    float currentz = origin[2] + spacing[2] * zindex;

    bool inBounds = false;
    size_t k = 0;

    while(!inBounds && k < totalNumFIDs)
    {
      float xmin = m_BoxCenter[3 * k] - m_BoxDims[3 * k] / 2.0;
      float xmax = m_BoxCenter[3 * k] + m_BoxDims[3 * k] / 2.0;
      float ymin = m_BoxCenter[3 * k + 1] - m_BoxDims[3 * k + 1] / 2.0;
      float ymax = m_BoxCenter[3 * k + 1] + m_BoxDims[3 * k + 1] / 2.0;
      float zmin = m_BoxCenter[3 * k + 2] - m_BoxDims[3 * k + 2] / 2.0;
      float zmax = m_BoxCenter[3 * k + 2] + m_BoxDims[3 * k + 2] / 2.0;
```
Move variable declarations inside the loop.